akka-actor-tests/src/test/scala/akka/pattern/BackoffOnRestartSupervisorSpec.scala

```diff
@@ -186,5 +191,27 @@ class BackoffOnRestartSupervisorSpec extends AkkaSpec with ImplicitSender {
         probe.expectTerminated(supervisor)
       }
     }
+
+    "respect withinTimeRange property of OneForOneStrategy" in {
+      val probe = TestProbe()
+      // The minimum backoff needs to be larger than "withinTimeRange" - otherwise it'll reset
+      val options = Backoff.onFailure(TestActor.props(probe.ref), "someChildName", 500 millis, 10 seconds, 0.0)
+        .withSupervisorStrategy(OneForOneStrategy(withinTimeRange = 300 millis) {
+          case _: TestActor.StoppingException ⇒ SupervisorStrategy.Stop
+        })
+      val supervisor = system.actorOf(BackoffSupervisor.props(options))
+      probe.expectMsg("STARTED")
+      filterException[TestActor.TestException] {
+        probe.watch(supervisor)
+        // Throw something immediately, we should get restarted
+        supervisor ! "THROW"
+        probe.expectMsg("STARTED")
+        // AFter waiting longer than the time range, throw again, we should not be restarted.
+        Thread.sleep(320)
+        supervisor ! "THROW"
+        // Superviser should've terminated.
```

Comment:
Ah, that makes sense, I just misinterpreted the meaning of that param - this way definitely makes more sense. Thanks